sounds.js

```diff
@@ -41,41 +44,181 @@ const soundFiles = {
+  const lBuffer = new Float32Array(bufferSize);
+  const rBuffer = new Float32Array(bufferSize);
+  
+  for (let i = 0; i < bufferSize; i++) {
+    const ratio = (bufferSize - i) / bufferSize; // will be 1 at the start of the loop and 0 at the end
+    const fadeAmount = easeOfNoise(ratio, 2);
+    lBuffer[i] = (1 - (2 * Math.random())) * fadeAmount;
+    rBuffer[i] = (1 - (2 * Math.random())) * fadeAmount;
+  }
+  
+  const buffer = audioContext.createBuffer(2, bufferSize, audioContext.sampleRate);
+  buffer.copyToChannel(lBuffer,0);
+  buffer.copyToChannel(rBuffer,1);
+
+  return buffer;
+
+}
+
+const upVectore = new THREE.Vector3(0, 1, 0);
+let reverbZonsPos = new THREE.Vector3(); 
+let cameraDirection = new THREE.Vector3();
+let localVector = new THREE.Vector3();
+const update = (timestamp, timeDiff) =>{
+  let inReverbZone = false;
+  // check whether the camera in the reverb zone 
+  for(const reverbZone of world.reverbZone){
+    reverbZonsPos.set(reverbZone[0], reverbZone[1], reverbZone[2]);
+    if(camera.position.distanceTo(reverbZonsPos) < reverbZone[3]){
+      inReverbZone = true;
+      break;
+    }
+  }
+  localVector.set(0, 0, -1);
+  cameraDirection = localVector.applyQuaternion( camera.quaternion );
+  cameraDirection.normalize();
+  // update listener 
+  const endTime = audioContext.currentTime + timeDiff;
+  if(audioContext.listener.positionX){
+			audioContext.listener.positionX.linearRampToValueAtTime( camera.position.x, endTime );
+			audioContext.listener.positionY.linearRampToValueAtTime( camera.position.y, endTime );
+			audioContext.listener.positionZ.linearRampToValueAtTime( camera.position.z, endTime );
+			audioContext.listener.forwardX.linearRampToValueAtTime( cameraDirection.x, endTime );
+			audioContext.listener.forwardY.linearRampToValueAtTime( cameraDirection.y, endTime );
+			audioContext.listener.forwardZ.linearRampToValueAtTime( cameraDirection.z, endTime );
+			audioContext.listener.upX.linearRampToValueAtTime( upVectore.x, endTime );
+			audioContext.listener.upY.linearRampToValueAtTime( upVectore.y, endTime );
+			audioContext.listener.upZ.linearRampToValueAtTime( upVectore.z, endTime );
+  }else{
+      audioContext.listener.setOrientation(cameraDirection.x, cameraDirection.y, cameraDirection.z, upVectore.x, upVectore.y, upVectore.z);
+      audioContext.listener.setPosition(camera.position.x, camera.position.y, camera.position.z);
+  }
+  
+  for(const sound of currentPlayingSound){
+    // update panner
+    if(sound.panner.positionX){
+      sound.panner.positionX.linearRampToValueAtTime(sound.voicer.position.x, endTime);
+			sound.panner.positionY.linearRampToValueAtTime(sound.voicer.position.y, endTime);
+			sound.panner.positionZ.linearRampToValueAtTime(sound.voicer.position.z, endTime);
+    }
+    else{
+      sound.panner.setPosition(sound.voicer.position.x, sound.voicer.position.y, sound.voicer.position.z);
+    }
+    
+    // handel reverb node

```

Comment:
Typo

Reply:
[typo and remove lerp](https://github.com/webaverse/app/pull/3056/commits/7168b3e52dc321e1d74aab271825acc8b2b39b6a)